letsencrypt-auto-source/letsencrypt-auto

```diff
@@ -613,9 +613,14 @@ if [ "$1" = "--le-auto-phase2" ]; then
     INSTALLED_VERSION="none"
   fi
   if [ "$LE_AUTO_VERSION" != "$INSTALLED_VERSION" ]; then
-    echo "Creating virtual environment..."
     DeterminePythonVersion
-    rm -rf "$VENV_PATH"
+    if [ -d "$VENV_PATH" ]; then
+        echo "Updating virtual environment..."
+        # The following will insure the latest Python core files and standard libraries (setuptools, pip) gets installed, while custom plugins and libraries installed in site-packages/ are kept as-is
+        rm -rf "$VENV_PATH"/bin/pip "$VENV_PATH"/bin/pip2 "$VENV_PATH"/bin/pip2.7 "$VENV_PATH"/bin/python "$VENV_PATH"/bin/python2 "$VENV_PATH"/bin/python2.7 "$VENV_PATH"/include/python2.7 "$VENV_PATH"/lib/python2.7/distutils "$VENV_PATH"/lib/python2.7/*-packages/easy_install.* "$VENV_PATH"/lib/python2.7/*-packages/pip "$VENV_PATH"/lib/python2.7/*-packages/pip-*.dist-info
```

Comment:
I think this approach will bite us in the future if the logic in virtualenv changes. Did the virtualenv authors make any promise that future versions will behave the same way in response to these deletions?

Also, we won't get the latest setuptools or pip in any case, as those are pinned in pipstrap. The safest path, it seems to me, is to leave the venv alone on upgrade.